First version of build script

angela-rph opened the pull request:
Script wrote in Python to be MultiOS
```
usage: builder.py [-h] [-a] [-b BOARD] [-i INO] [-s SKETCHES] 
Automatic build script 
optional arguments: 
  -h, --help            show this help message and exit 
  -a, --all             -a : automatic build - build all sketches for all 
                        board 
  -b BOARD, --board BOARD 
                        -b <board pattern>: pattern to find one or more boards 
                        to build 
  -i INO, --ino INO     -i <ino filepath>: single ino file to build 
  -s SKETCHES, --sketches SKETCHES 
                        -s <sketch pattern>: pattern to find one or more 
                        sketch to build

```

angela-rph committed:
First version of build script

Python / MultiOS

Signed-off-by: Angela Randolph <angela.randolph@st.com>

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+#Common path
+arduino_path = config["ARDUINO_PATH"]
+arduino_packages = config["ARDUINO_PACKAGES"]
+build_output_dir=config["BUILD_OUPUT_DIR"]
```
`build_output_dir` have to be unique.
If several users launch a build on the same machine this will not work properly if they kept the default build output dirname in the `config.json` as _arduino-builder_ will use the same output dir.
Moreover, it will probably fine to remove it at the end of script.
Maybe a good solution should be to decorate this name if default name is kept.

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+
+assert os.path.exists(arduino_path), 'Path does not exist: %s . Please set this path in the json config file' %arduino_path
+assert os.path.exists(arduino_packages), 'Path does not exist: %s . Please set this path in the json config file' %arduino_packages
+assert os.path.exists(build_output_dir), 'Path does not exist: %s . Please set this path in the json config file' %build_output_dir
```
This one is created by the script, error is probably not linked to config file except if this one is empty.
Anyway check is already done in `createFolder` which try to create folder if not excipt else raised an error.
So, maybe the right assertion to do is to check if it is not empty.

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+assert os.path.exists(arduino_path), 'Path does not exist: %s . Please set this path in the json config file' %arduino_path
+assert os.path.exists(arduino_packages), 'Path does not exist: %s . Please set this path in the json config file' %arduino_packages
+assert os.path.exists(build_output_dir), 'Path does not exist: %s . Please set this path in the json config file' %build_output_dir
+assert os.path.exists(output_dir), 'Path does not exist: %s . Please set this path in the json config file' %output_dir
```
ditto

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+createFolder(bin_dir)
+createFolder(std_dir)
+
+assert os.path.exists(arduino_path), 'Path does not exist: %s . Please set this path in the json config file' %arduino_path
```
Following assertions should be raised before trying to create the folder.
Just after init them:
https://github.com/fpistm/stm32_tools/pull/3/files#diff-1f5ada8250525fa6f0fcb5f0e69dc739R59

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+sketches = find_inos(args)
+variants = find_board(args)
+
+if len(sys.argv)<2 : #Si aucun args
```
English please ;)

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+        print("Platform is Windows")
+        arduino_path = 'C:\\Program Files (x86)\\Arduino' #arduino default path
+        arduino_packages = home+'\\AppData\\Local\\Arduino15\\packages' #Windows 7
+        build_output_dir = tempdir+'\\temp_arduinoBuilderOutput' #Windows 7 \\temp_build_output' #temporary directory using by arduino builder
```
Comment not really clear

fpistm commented on arduino-builder/builder.py:
```diff
@@ -0,0 +1,280 @@
+    print("\n****************** PROCESSING COMPLETED ******************")
+    print("PASSED = {}/{}".format(nb_build_passed,nb_build_total))
+    print("FAILED = {}/{}".format(nb_build_failed,nb_build_total))
+    print("Logs are available at", output_dir)
```
Please print as string not as list

fpistm reviewed (dismissed):
Anyway nice first version

fpistm committed:
Rename builder.py to arduino-builder.py

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Use Black code formatter

Black is the uncompromising Python code formatter:
https://github.com/ambv/black/

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Use flake8 for style enforcement

Flake8 is a tool for style guide enforcement
https://github.com/PyCQA/flake8

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Update result format

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Review sketches list management

Find all inos only when all or sketch pattern options are used.
Single ino and pattern are mutually exclusive.

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Add sketches list file option

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Review boards list management

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Reorder functions

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Rename some functions

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Remove usb variant parameter

usb menu is not available for all variant.
Define it for a variant which do not have it produce a huge binary.
(ex: Blink for BluePill is twice as big)

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Enable all warning abd make verbose optional

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Reduce script output size

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Use unique build id

Allow to launch multiple build on the same host

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Review outputs

Update output tree structure to be more efficient
Update output files format

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

fpistm committed:
[arduino-builder] Add some clean capabilities

Delete build output directory.
Add new option to clean build result directory.

Signed-off-by: Frederic Pillon <frederic.pillon@st.com>

angela-rph committed:
Merge pull request #1 from fpistm/review_buildscript

Review buildscript

angela-rph committed:
Add exclude list file option
Rework on print function

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Add file header comments

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Rework on arduino builder command

Add new "libraries" argument
Review user output (add statistic)

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Rework @manage_exclude_list

Review header file comment

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Code formatting with Black

Signed-off-by: Angela Randolph <angela.randolph@st.com>

fpistm committed:
Fix bugs

Signed-off-by: Frederic.Pillon <frederic.pillon@st.com>
Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Update @find_board

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Rework on binaries copy.

Copy ino binairies only when "binaries" options is used.

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Add travis options

If build failed (required travis option) :
 - display the contents of stderr file
 - exit from python


Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Code formatting with Black

Signed-off-by: Angela Randolph <angela.randolph@st.com>

angela-rph committed:
Fix bugs

Signed-off-by: Angela Randolph <angela.randolph@st.com>
